pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -465,24 +580,44 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 	} else {
 		postgresResourceRequirements := getPostgresResourceRequirements(instance)
 
-		containers := foundPostgresqlDeploymentConfig.Spec.Template.Spec.Containers
+		containers := foundPostgresqlDeployment.Spec.Template.Spec.Containers
 		for i := range containers {
 			if containers[i].Name == cfg.PostgresContainerName {
 				if reflect.DeepEqual(containers[i].Resources, postgresResourceRequirements) == false {
-					reqLogger.Info("Postgres container resource requirements are different than in the UnifiedPushServer spec or the operator defaults", "DeploymentConfig.Namespace", foundPostgresqlDeploymentConfig.Namespace, "DeploymentConfig.Name", foundPostgresqlDeploymentConfig.Name, "Found resource requirements", containers[i].Resources, "Spec resource requirements", postgresResourceRequirements)
+					reqLogger.Info("Postgres container resource requirements are different than in the UnifiedPushServer spec or the operator defaults", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "Found resource requirements", containers[i].Resources, "Spec resource requirements", postgresResourceRequirements)
 
 					containers[i].Resources = postgresResourceRequirements
 
 					// enqueue
-					err = operatorClient.Update(context.TODO(), foundPostgresqlDeploymentConfig)
+					err = operatorClient.Update(context.TODO(), foundPostgresqlDeployment)
 					if err != nil {
-						reqLogger.Error(err, "Failed to update DeploymentConfig", "DeploymentConfig.Namespace", foundPostgresqlDeploymentConfig.Namespace, "DeploymentConfig.Name", foundPostgresqlDeploymentConfig.Name)
+						reqLogger.Error(err, "Failed to update Deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name)
 						return reconcile.Result{}, err
 					}
 					return reconcile.Result{Requeue: true}, nil
 				}
 			}
 		}
+
+		desiredImage := constants.PostgresImage
+
+		containerSpec := findContainerSpec(foundPostgresqlDeployment, cfg.PostgresContainerName)
+		if containerSpec == nil {
+			reqLogger.Info("Skipping image reconcile: Unable to find container spec in deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", cfg.PostgresContainerName)
+		} else if containerSpec.Image != desiredImage {
+			reqLogger.Info("Container spec in deployment is using a different image. Going to update it now.", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", cfg.PostgresContainerName, "ExistingImage", containerSpec.Image, "DesiredImage", desiredImage)
+
+			// update
+			updateContainerSpecImage(foundPostgresqlDeployment, cfg.PostgresContainerName, desiredImage)
+
+			// enqueue
+			err = operatorClient.Update(context.TODO(), foundPostgresqlDeployment)
```

Comment:
When you rebase on master now that #78 is merged, lines like this one will have to change a little.